### ISSUE TITLE
Delete customization not 100%

### ISSUE THREAD

**Issue by erapade** [mampfes/hacs_waste_collection_schedule#3118]
### I Have A Problem With:

Sensor configuration

### What's Your Problem

It was a bit hard to understand how everything works so I started to monitor the JSON file in .store and realized that only the alias was removed when deleting/removing a customization. Also, in the GUI, the child window is not closed after selecting delete and then submit. All this causes confusion for new users

### Checklist Source Error

- [ ] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [ ] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
Thanks for bringing this up that's not how this should behave.


> in the GUI, the child window is not closed after selecting delete and then submit. All this causes confusion for new users

I do not like this either, but there is no way to add custom buttons in this config_flow. There is only the submit button.

**Comment by bbr111** [comment]
Agreed – from a UX perspective the dialog should be closed or the flow aborted after deletion.
Unfortunately this is a limitation of Home Assistant’s config_flow (no custom buttons, submit-only), so this can’t be improved further on our side at the moment.
